### PR TITLE
Submit dev.lizardbyte.app.Sunshine

### PR DESCRIPTION
**Pull request by istori1**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: dev.lizardbyte.app.sunshine.desktop [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+
+Name=Sunshine
+Comment=Self-hosted game stream host for Moonlight. 
+Categories=AudioVideo;
+
+Icon=dev.lizardbyte.app.sunshine
+Exec=
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.desktop:10]: this should be set.

File: dev.lizardbyte.app.sunshine.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          project-id: 21814
+          stable-only: true
+          url-template: https://github.com/Intel-Media-SDK/MediaSDK/archive/refs/tags/intel-mediasdk-$version.tar.gz
+
+  - name: cuda
+    disabled: false
+    buildsystem: simple
+    only-arches:
+      - x86_64
+      - aarch64
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:149]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  - name: sunshine
+    disabled: false
+    buildsystem: cmake
+    builddir: true
+    build-options:
+      append-path: /usr/lib/sdk/node18/bin
+      cxxflags: -I${C_INCLUDE_PATH}/libevdev-1.0
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_PREFIX=/app
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:181]: this is set automatically
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      - type: file
+        only-arches:
+          - aarch64
+        url: https://developer.download.nvidia.com/compute/cuda/12.1.1/local_installers/cuda_12.1.1_530.30.02_linux_sbsa.run
+        sha256: 45ea4cd860f0a26d3db8ce032530f2ee0b55abdd587545213d395a73623b4278
+        dest-filename: cuda.run
+
+  - name: sunshine
+    disabled: false
+    buildsystem: cmake
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:174]: cmake-ninja?

File: dev.lizardbyte.app.sunshine.yml [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        dest-filename: cuda.run
+      - type: file
+        only-arches:
+          - aarch64
+        url: https://developer.download.nvidia.com/compute/cuda/12.1.1/local_installers/cuda_12.1.1_530.30.02_linux_sbsa.run
+        sha256: 45ea4cd860f0a26d3db8ce032530f2ee0b55abdd587545213d395a73623b4278
+        dest-filename: cuda.run
+
+  - name: sunshine
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:173]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+finish-args:
+  - --device=all
+  - --env=PULSE_PROP_media.category=Manager
+  - --persist=.config/sunshine
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=x11
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:18]: fallback-x11?

File: dev.lizardbyte.app.sunshine.yml [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+        url: https://github.com/numactl/numactl/archive/refs/tags/v2.0.16.tar.gz
+        sha256: a35c3bdb3efab5c65927e0de5703227760b1101f5e27ab741d8f32b3d5f0a44c
+        x-checker-data:
+          type: anitya
+          project-id: 2507
+          stable-only: true
+          url-template: https://github.com/numactl/numactl/archive/refs/tags/v$version.tar.gz
+
+  - name: intel-mediasdk
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:126]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        url: https://www.freedesktop.org/software/libevdev/libevdev-1.13.0.tar.xz
+        sha256: 9edf2006cc86a5055279647c38ec923d11a821ee4dc2c3033e8d20e8ee237cd9
+        x-checker-data:
+          type: anitya
+          project-id: 20540
+          stable-only: true
+          url-template: https://www.freedesktop.org/software/libevdev/libevdev-$version.tar.xz
+
+  - name: numactl
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:109]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    buildsystem: autotools
+    make-args:
+      - install
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:112]:
```suggestion
```

make install is called.

File: dev.lizardbyte.app.sunshine.yml [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+        url: https://src.fedoraproject.org/rpms/avahi.git
+        commit: e8ad0f7bbb7fdf82a0f0be128bab07b5928322f0
+        dest: rpms
+      - type: shell
+        commands:
+          - cat rpms/*.patch | patch -Np1 -d .
+          - autoreconf -ivf
+
+  - name: libevdev
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:91]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+        url: https://github.com/boostorg/boost/releases/download/boost-1.82.0/boost-1.82.0.tar.xz
+        sha256: fd60da30be908eff945735ac7d4d9addc7f7725b1ff6fcdcaede5262d511d21e
+        x-checker-data:
+          type: anitya
+          project-id: 6845
+          stable-only: true
+          url-template: https://github.com/boostorg/boost/releases/download/boost-$version/boost-$version.tar.xz
+
+  - name: avahi
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:49]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+  - /lib/cmake
+  - /lib/pkgconfig
+  - /lib/*.la
+  - /lib/*.a
+  - /share/man
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+  - name: boost
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:33]:
```suggestion
```

File: dev.lizardbyte.app.sunshine.yml [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+        url: https://github.com/Intel-Media-SDK/MediaSDK/archive/refs/tags/intel-mediasdk-23.2.0.tar.gz
+        sha256: 6b33ec917ef0c6395fb66c7b93aae4ae31b3d1dedb5d7d39e0c69f45acb7c4fb
+        x-checker-data:
+          type: anitya
+          project-id: 21814
+          stable-only: true
+          url-template: https://github.com/Intel-Media-SDK/MediaSDK/archive/refs/tags/intel-mediasdk-$version.tar.gz
+
+  - name: cuda
+    disabled: false
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:145]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build dev.lizardbyte.app.sunshine

File: spawn-host.patch [hunk 14]
```diff
@@ -0,0 +1,21 @@
+--- ~/src_assets/linux/assets/apps.json
++++ ~/src_assets/linux/assets/apps.json
+@@ -12,15 +12,15 @@
+       "image-path": "desktop.png",
+       "prep-cmd": [
+         {
+-          "do": "xrandr --output HDMI-1 --mode 1920x1080",
+-          "undo": "xrandr --output HDMI-1 --mode 1920x1200"
++          "do": "flatpak-spawn --host xrandr --output HDMI-1 --mode 1920x1080",
++          "undo": "flatpak-spawn --host xrandr --output HDMI-1 --mode 1920x1200"
```
Comment (hfiguiere) [inline spawn-host.patch:10]: instead you should ship xrandr. But how does this work with wayland?

File: spawn-host.patch [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+         }
+       ]
+     },
+     {
+       "name": "Steam Big Picture",
+       "detached": [
+-        "setsid steam steam://open/bigpicture"
++        "flatpak-spawn --host setsid steam steam://open/bigpicture"
```
Comment (hfiguiere) [inline spawn-host.patch:18]: so this require something on the host? What's the failure mode?

File: dev.lizardbyte.app.sunshine.yml [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+    builddir: true
+    build-options:
+      append-path: /usr/lib/sdk/node18/bin
+      cxxflags: -I${C_INCLUDE_PATH}/libevdev-1.0
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_PREFIX=/app
+      - -DCMAKE_CUDA_COMPILER=/app/cuda/bin/nvcc
+      - -DSUNSHINE_ASSETS_DIR=share/sunshine
+      - -DSUNSHINE_EXECUTABLE_PATH="flatpak run dev.lizardbyte.sunshine"
```
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:184]: not sure what's this about.

File: dev.lizardbyte.app.sunshine.yml (hunk 6 above)
Comment (istori1) [reply dev.lizardbyte.app.sunshine.yml:18]: Needs XWayland for mouse position on the screen.

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (istori1) [reply dev.lizardbyte.app.sunshine.yml:184]: -DCMAKE_CUDA_COMPILER=/app/cuda/bin/nvcc is needed for finding the CUDA compiler

Custom CMake variables of the project
-DSUNSHINE_ASSETS_DIR=share/sunshine
-DSUNSHINE_EXECUTABLE_PATH="flatpak run dev.lizardbyte.sunshine"

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.sunshine

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (hfiguiere) [reply dev.lizardbyte.app.sunshine.yml:182]: systemd? There is no systemd.

Comment (hfiguiere) [reply dev.lizardbyte.app.sunshine.yml:182]: I only care about line 182 here.

**Comment by hfiguiere** [comment]
Build system fail to honour prefix

```
-- Installing: /usr/share/icons/sunshine.svg
CMake Error at cmake_install.cmake:158 (file):
  file INSTALL cannot copy file "/run/build/sunshine/sunshine.svg" to
  "/usr/share/icons/sunshine.svg": Read-only file system.
```

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

File: dev.lizardbyte.app.Sunshine.yml [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+    buildsystem: cmake
+    builddir: true
+    build-options:
+      append-path: /usr/lib/sdk/node18/bin
+      cxxflags: -I${C_INCLUDE_PATH}/libevdev-1.0
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_CUDA_COMPILER=/app/cuda/bin/nvcc
+      - -DSUNSHINE_ASSETS_DIR=share/sunshine
+      - -DSUNSHINE_EXECUTABLE_PATH="flatpak run --command=sunshine dev.lizardbyte.app.Sunshine"
```
Comment (hfiguiere) [inline dev.lizardbyte.app.Sunshine.yml:182]: I will reiterate. flatpak do NO exist in the sandbox. That mean THIS IS WRONG. I still haven't explained what it is for (there is no systemd either).

File: dev.lizardbyte.app.Sunshine.yml [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+      - -DSUNSHINE_ENABLE_DRM=ON
+      - -DSUNSHINE_ENABLE_CUDA=ON
+      - -DSUNSHINE_CONFIGURE_FLATPAK=ON
+    post-install:
+      - install -D $FLATPAK_BUILDER_BUILDDIR/packaging/linux/flatpak/scripts/* $FLATPAK_DEST/bin
+      # - install -D $FLATPAK_BUILDER_BUILDDIR/sunshine.svg $FLATPAK_DEST/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg
+      - install -D $FLATPAK_BUILDER_BUILDDIR/sunshine.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -D $FLATPAK_BUILDER_BUILDDIR/$FLATPAK_ID.sh $FLATPAK_DEST/bin/$FLATPAK_ID.sh
+      - install -D $FLATPAK_BUILDER_BUILDDIR/$FLATPAK_ID.metainfo.xml $FLATPAK_DEST/share/metainfo/$FLATPAK_ID.metainfo.xml
+      - install -D $FLATPAK_BUILDER_BUILDDIR/$FLATPAK_ID.desktop $FLATPAK_DEST/share/applications/$FLATPAK_ID.desktop
```
Comment (hfiguiere) [inline dev.lizardbyte.app.Sunshine.yml:194]:
```suggestion
      - install -D ../packaging/linux/flatpak/scripts/* $FLATPAK_DEST/bin
      # - install -D ../sunshine.svg $FLATPAK_DEST/share/icons/hicolor/scalable/apps/$FLATPAK_ID.svg
      - install -D ../sunshine.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
      - install -D ../$FLATPAK_ID.sh $FLATPAK_DEST/bin/$FLATPAK_ID.sh
      - install -D ../$FLATPAK_ID.metainfo.xml $FLATPAK_DEST/share/metainfo/$FLATPAK_ID.metainfo.xml
      - install -D ../$FLATPAK_ID.desktop $FLATPAK_DEST/share/applications/$FLATPAK_ID.desktop
```

File: dev.lizardbyte.app.sunshine.yml
Comment (hfiguiere) [inline dev.lizardbyte.app.sunshine.yml:1]: this file is duplicated with an incorrect app-id judging from the rest.

File: prefix.patch
Comment (hfiguiere) [inline prefix.patch:1]: you should submit this patch upstream.

**Review by hfiguiere — COMMENTED** [review]
have the appstream and desktop files been submitted upstream?

File: prefix.patch
Comment (istori1) [reply prefix.patch:1]: Submitted https://github.com/LizardByte/Sunshine/pull/1208

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (ReenigneArcher) [reply dev.lizardbyte.app.sunshine.yml]: `-DSUNSHINE_EXECUTABLE_PATH` is used in our source code. If this is how sunshine is launched when running the flatpak, then it is correct.

File: dev.lizardbyte.app.Sunshine.desktop [hunk 19]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+
+Name=Sunshine
+Comment=Self-hosted game stream host for Moonlight. 
+Categories=AudioVideo;
+
+Icon=dev.lizardbyte.app.sunshine
```
Comment (hfiguiere) [inline dev.lizardbyte.app.Sunshine.desktop:9]: you changed the ID but not here.

Comment (hfiguiere) [reply dev.lizardbyte.app.Sunshine.desktop:9]: Nor just below.

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (hfiguiere) [reply dev.lizardbyte.app.sunshine.yml]: and how does that work?

File: dev.lizardbyte.app.Sunshine.yml (hunk 17 above)
Comment (istori1) [reply dev.lizardbyte.app.Sunshine.yml:182]: This is to give the option to configure as systemd user service . The user would copy it to ~/.config/systemd/user/ and enable it as a service.

Comment (istori1) [reply dev.lizardbyte.app.Sunshine.yml:182]: Result is
```
[Unit]
Description=@PROJECT_DESCRIPTION@
StartLimitIntervalSec=500
StartLimitBurst=5
PartOf=graphical-session.target
Wants=xdg-desktop-autostart.target
After=xdg-desktop-autostart.target

[Service]
ExecStart=flatpak run --command=sunshine dev.lizardbyte.app.Sunshine
Restart=on-failure
RestartSec=5s

[Install]
WantedBy=xdg-desktop-autostart.target
```

File: dev.lizardbyte.app.Sunshine.desktop (hunk 19 above)
Comment (istori1) [reply dev.lizardbyte.app.Sunshine.desktop]: Missed this. Thank you

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

File: spawn-host.patch (hunk 15 above)
Comment (ReenigneArcher) [reply spawn-host.patch:8]: In upstream we patch it with `SED`, which allows us to add/remove apps without worrying about updating a patch file.

https://github.com/LizardByte/Sunshine/blob/master/packaging/linux/flatpak/dev.lizardbyte.sunshine.yml#L233-L242

I understand `SED` can be confusing if you don't understand regex.

I don't use flatpak, but don't the `do/undo` commands also need to be patched? https://github.com/LizardByte/Sunshine/blob/master/src_assets/linux/assets/apps.json#L15-L16

> What's the failure mode?
Steam won't launch.

Comment (hfiguiere) [reply spawn-host.patch:8]: sed as a failure mode that is not a failure. So things silently fail and then use complain things are broken.

Maybe you like that kind of failure mode.

Comment (hfiguiere) [reply spawn-host.patch:8]: as for steam, what if the user has it as a flatpak?

File: dev.lizardbyte.app.Sunshine.yml (hunk 17 above)
Comment (hfiguiere) [reply dev.lizardbyte.app.Sunshine.yml:182]: not with the current sandbox permissions.

Comment (hfiguiere) [reply dev.lizardbyte.app.Sunshine.yml:182]: also this is not the way auto starting in the user session is handled.

File: dev.lizardbyte.app.sunshine.desktop (hunk 1 above)
Comment (ReenigneArcher) [reply dev.lizardbyte.app.sunshine.desktop]: So... our upstream desktop file is currently only used for our AppImage. These fields are populated using a the configure option in cmake. Changes to upstream regarding this should be handled in the CMakeLists so the file is configured in a similar manner for flatpak.

Example:
https://github.com/LizardByte/Sunshine/blob/master/CMakeLists.txt#L85-L86

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (ReenigneArcher) [reply dev.lizardbyte.app.sunshine.yml]: Sorry, I was mistaken, currently it's only used in the service file. So however this is set is what is used when cmake configures the service file.

In the future we may use this value for other functions in the code... for example we are contemplating a method to restart our application on demand, so this variable would work perfectly for that.

File: spawn-host.patch (hunk 15 above)
Comment (ReenigneArcher) [reply spawn-host.patch:8]: > as for steam, what if the user has it as a flatpak?

Then they can manually change it in our config UI (through web browser).

File: dev.lizardbyte.app.sunshine.yml (hunk 4 above)
Comment (ReenigneArcher) [reply dev.lizardbyte.app.sunshine.yml]: Could you explain the benefit of `cmake-ninja` versus `cmake`? I'm not familiar with `cmake-ninja` so would just like to learn if you don't mind explaining it.

File: dev.lizardbyte.app.sunshine.yml (hunk 16 above)
Comment (hfiguiere) [reply dev.lizardbyte.app.sunshine.yml]: > In the future we may use this value for other functions in the code... for example we are contemplating a method to restart our application on demand, so this variable would work perfectly for that.

actually with this value it wouldn't work. To restart yourself in the sandbox, just exec yourself. It's not different.

File: dev.lizardbyte.app.sunshine.yml (hunk 4 above)
Comment (hfiguiere) [reply dev.lizardbyte.app.sunshine.yml]: it faster.

It's `cmake -G Ninja`

File: dev.lizardbyte.app.Sunshine.metainfo.xml [hunk 20]
```diff
@@ -14,6 +14,10 @@
     <control>touch</control>
     <control>gamepad</control>
   </supports>
+
+  <url type="homepage">https://github.com/LizardByte/Sunshine</url>
```
Comment (ReenigneArcher) [inline dev.lizardbyte.app.Sunshine.metainfo.xml:18]:
```suggestion
  <url type="homepage">https://app.lizardbyte.dev</url>
```

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
> have the appstream and desktop files been submitted upstream?

Submission is planned.

File: dev.lizardbyte.app.sunshine.yml (hunk 4 above)
Comment (istori1) [reply dev.lizardbyte.app.sunshine.yml]: Build fails with `cmake-ninja`

File: spawn-host.patch (hunk 14 above)
Comment (istori1) [reply spawn-host.patch]: The user has several options to choose (Wayland compatible) this one is X11 specific. This part of the patch is no longer needed by including xrandr with build

File: dev.lizardbyte.app.Sunshine.yml [hunk 21]
```diff
@@ -0,0 +1,217 @@
...
+sdk: org.freedesktop.Sdk
+separate-locales: false
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+command: dev.lizardbyte.app.Sunshine.sh
+
+finish-args:
+  - --device=all
+  - --env=PULSE_PROP_media.category=Manager
+  - --filesystem=~/.config/systemd/user:create
```
Comment (hfiguiere) [inline dev.lizardbyte.app.Sunshine.yml:13]: no, not really. Why not removing the sandbox then.

File: dev.lizardbyte.app.Sunshine.yml [hunk 22]
```diff
@@ -0,0 +1,217 @@
...
+  - --persist=.config/sunshine
```
Comment (hfiguiere) [inline dev.lizardbyte.app.Sunshine.yml:14]: it doesn't use the standard directories?

Comment (istori1) [reply dev.lizardbyte.app.Sunshine.yml]: It does not follow the standard directories (XDG).

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

File: dev.lizardbyte.app.sunshine.yml (hunk 4 above)
Comment (ReenigneArcher) [reply dev.lizardbyte.app.sunshine.yml]: We can build most of our packages with `cmake -G Ninja`, so not sure why it fails in the flatpak build. Also cannot build in our repo by changing our flatpak manifest to `cmake-ninja`.

**Comment by barthalion** [comment]
The log says

```
[2023:05:04:20:51:40]: Info: System tray created
```

but I don't see this actually happening on GNOME with the relevant tray extension.

Then, there's this:

```
2023:05:04:20:52:56]: Info: Found display [wayland-0]
[2023:05:04:20:52:56]: Info: Found interface: zxdg_output_manager_v1(5) version 3
[2023:05:04:20:52:56]: Info: Found interface: wl_output(33) version 3
[2023:05:04:20:52:56]: Info: Offset: 0x0
[2023:05:04:20:52:56]: Info: Resolution: 2560x1440
[2023:05:04:20:52:56]: Info: Name: HDMI-1
[2023:05:04:20:52:56]: Info: Found monitor: Dell Inc. 27"
[2023:05:04:20:52:56]: Info: HDMI-1: Dell Inc. 27"
[2023:05:04:20:52:56]: Warning: Mismatch on expected Resolution compared to actual resolution: 0x0 vs 2560x1440
[2023:05:04:20:52:56]: Error: Unable to initialize capture method
[2023:05:04:20:52:56]: Error: Platform failed to initialize
[2023:05:04:20:52:56]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2023:05:04:20:52:56]: Info: Trying encoder [nvenc]
[2023:05:04:20:52:56]: Info: System tray created

(sunshine:5): Gtk-CRITICAL **: 20:52:56.376: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
[2023:05:04:20:52:57]: Info: Encoder [nvenc] failed
[2023:05:04:20:52:57]: Info: Trying encoder [vaapi]
[2023:05:04:20:52:59]: Info: Encoder [vaapi] failed
[2023:05:04:20:52:59]: Info: Trying encoder [software]
[2023:05:04:20:53:01]: Info: Encoder [software] failed
[2023:05:04:20:53:01]: Fatal: Couldn't find any working encoder
[2023:05:04:20:53:01]: Error: Video failed to initialize
(...)
[2023:05:04:20:57:57]: Info: Executing Do Cmd: [xrandr --output HDMI-1 --mode 1920x1080]
[2023:05:04:20:57:57]: Warning: run_unprivileged() is not yet implemented for this platform. The new process will run with Sunshine's permissions.
[2023:05:04:20:57:57]: Info: Executing [Desktop]
[2023:05:04:20:57:57]: Info: CLIENT CONNECTED
/usr/include/c++/12.2.0/bits/stl_vector.h:1206: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::front() [with _Tp = video::encoder_t; _Alloc = std::allocator<video::encoder_t>; reference = video::encoder_t&]: Assertion '!this->empty()' failed.
/app/bin/dev.lizardbyte.app.Sunshine.sh: line 10:     5 Aborted                 (core dumped) sunshine
```

Does it actually work on Wayland?

Finally, it seems to open the web UI before it's actually ready to serve traffic. I had to refresh the tab manually after a few seconds to get to the UI.

**Comment by hfiguiere** [comment]
that assertion is a classic since Flaptak, unlike other, still build with the C++ assertions on. And this is a bug in upstream code.

**Comment by istori1** [comment]
For Wayland, it uses either a wlroots specific API or KMS Grab. With KMS all Wayland compositors should work, but that requires root. `sudo -i PULSE_SERVER=unix:$(pactl info | awk '/Server String/{print$3}') flatpak run dev.lizardbyte.app.Sunshine`

**Comment by hfiguiere** [comment]
> For Wayland, it uses either a wlroots specific API or KMS Grab. With KMS all Wayland compositors should work, but that requires root. `sudo -i PULSE_SERVER=unix:$(pactl info | awk '/Server String/{print$3}') flatpak run dev.lizardbyte.app.Sunshine`

If thats what the user has to do to run it, then it's broken.

**Comment by barthalion** [comment]
Yeah, I don't see the point of granting Wayland access out of the box if it doesn't actually work.

**Comment by istori1** [comment]
Existing users already use Sunshine on Wayland with its current limitations. We want to keep Wayland available. A note can be added in the description for limited Wayland support and recommending using Sunshine on X11.

**Comment by hfiguiere** [comment]
What @barthalion said. If the user have to disable it to get it to work then that mean it's not ready.

**Comment by barthalion** [comment]
My point is that you already need to have it documented somewhere that users are supposed to execute a custom command with `sudo`. Just making sure it's mentioned to pass `--socket=wayland` to `flatpak run` is not changing anything from the practical standpoint. And the bottom line is that it doesn't work on Wayland out of the box.

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, stop

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Comment by istori1** [comment]
bot, build dev.lizardbyte.app.Sunshine

**Pull request closed without merging**